### PR TITLE
Support keypoint names metadata for pose models

### PR DESCRIPTION
**Pull request by Y-T-G**
```python
from ultralytics import YOLO
model = YOLO("yolo11n-pose.pt")
results = model.train(data="coco8-pose.yaml", epochs=1, exist_ok=True)
model = YOLO("runs/pose/train/weights/best.pt")
print(model.kpt_names)
results = model()
for i, result in enumerate(results):
    classes = result.boxes.cls.tolist()
    kpts = result.keypoints.xy.tolist()
    for cls, kpt in zip(classes, kpts):
        print(dict(zip(model.kpt_names[cls], kpt)))
```

```python
{'nose': [652.0538940429688, 358.3094482421875], 'left_eye': [659.1234130859375, 331.9946594238281], 'right_eye': [627.3772583007812, 326.6024169921875], 'left_ear': [617.7488403320312, 332.4612731933594], 'right_ear': [528.3828735351562, 317.5885009765625], 'left_shoulder': [631.0384521484375, 426.4150390625], 'right_shoulder': [349.49560546875, 419.6319580078125], 'left_elbow': [826.8046875, 508.06060791015625], 'right_elbow': [211.17703247070312, 594.3692016601562], 'left_wrist': [969.5433349609375, 545.4529418945312], 'right_wrist': [307.29205322265625, 714.197021484375], 'left_hip': [497.0877380371094, 711.0231323242188], 'right_hip': [317.22705078125, 717.8145751953125], 'left_knee': [506.6791076660156, 596.5565795898438], 'right_knee': [265.020263671875, 610.9822998046875], 'left_ankle': [509.8736267089844, 711.9202880859375], 'right_ankle': [296.5079650878906, 720.0]}
{'nose': [999.6265258789062, 202.27635192871094], 'left_eye': [1026.495849609375, 172.95150756835938], 'right_eye': [964.1179809570312, 168.3924102783203], 'left_ear': [1052.182861328125, 186.03170776367188], 'right_ear': [905.0728759765625, 177.27322387695312], 'left_shoulder': [1048.4482421875, 337.449462890625], 'right_shoulder': [853.6993408203125, 327.23736572265625], 'left_elbow': [1064.6339111328125, 519.8368530273438], 'right_elbow': [784.0687255859375, 523.6309814453125], 'left_wrist': [1090.9337158203125, 486.72943115234375], 'right_wrist': [960.6470336914062, 459.84271240234375], 'left_hip': [1034.6888427734375, 720.0], 'right_hip': [900.2042846679688, 720.0], 'left_knee': [1074.7962646484375, 720.0], 'right_knee': [919.947509765625, 720.0], 'left_ankle': [1072.2864990234375, 720.0], 'right_ankle': [894.8368530273438, 720.0]
```

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Adds human-readable keypoint names to pose datasets and propagates them through training, export, and inference for clearer, more informative pose outputs. 🧍‍♂️🐶✋

### 📊 Key Changes
- Dataset updates: Added `kpt_names` to pose YAMLs:
  - `coco-pose.yaml`, `coco8-pose.yaml`, `dog-pose.yaml`, `hand-keypoints.yaml`, `tiger-pose.yaml` now include per-class keypoint name lists.
- Training: `PoseModel` now sets `model.kpt_names` from dataset config, with an automatic fallback to numeric labels if none provided.
- Export: Pose model exports now include `kpt_names` in model metadata for downstream consumers.
- Inference backend: `AutoBackend` now reads `kpt_names` from exported metadata, ensuring keypoint names are available across formats/backends.

### 🎯 Purpose & Impact
- Improved clarity: Users can see meaningful keypoint names (e.g., “left_wrist”, “nose”) instead of indices, aiding interpretation and debugging. 🔎
- Better visualization and reporting: Plots, logs, and UIs can show named keypoints for more readable outputs.
- Seamless deployment: Named keypoints are preserved through export and inference, improving interoperability across runtimes. 🚀
- Backward compatible: If `kpt_names` aren’t provided, defaults to numeric labels, so existing workflows remain unaffected.
- Multi-class readiness: Supports per-class keypoint naming, enabling more complex pose datasets.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated response to help streamline reviews. An Ultralytics engineer will also assist you here shortly 🙌

- ✅ Define a Purpose: Clearly explain the purpose of your fix or feature in your PR description, and link to any relevant issues using the Ultralytics Issues page: [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ Synchronize with Source: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ Ensure CI Checks Pass: Verify all Ultralytics CI checks are passing here: [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/). If any checks fail, please address the issues.
- ✅ Update Documentation: Update the relevant project docs for any new or modified features, including usage guidance for keypoint name metadata in pose workflows: [documentation](https://docs.ultralytics.com/).
- ✅ Add Tests: If applicable, include or update tests to cover your changes, e.g., round-trip export/import of `kpt_names`, default handling when `kpt_names` is absent, and backend loading (including MNN). Confirm that all tests are passing.
- ✅ Sign the CLA: Please ensure you have signed our Contributor License Agreement if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message: [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/).
- ✅ Minimize Changes: Limit your changes to the minimum necessary for your feature addition. "It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is." — Bruce Lee

Additional notes for this PR:
- For dataset updates (e.g., adding `kpt_names` to `coco8-pose.yaml`), please ensure consistent formatting and backward compatibility where `kpt_names` may be missing.
- For exporter and backend changes, ensure metadata keys are consistently serialized/deserialized and safely handled across tasks that don’t use pose.
- If you’ve added new user-facing behavior, consider adding a brief usage snippet to docs and tests to validate expected outputs during inference and export/import cycles.

For more guidance, please refer to our Contributing Guide: [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🎯🚀

**Comment by glenn-jocher** [comment]
@Y-T-G nice, this looks really good!

@RizwanMunawar please double check the keypoint names here.

**Comment by Y-T-G** [comment]
@glenn-jocher @RizwanMunawar For tiger-pose and dog-pose, I couldn't find the keypoint names in docs. I just manually drew the keypoints on the images and based it on that. Also noticed that dog-pose is missing two keypoints every one keypoint, which is why the names for those are set to `None` because I don't have a way to tell which keypoints those missing ones are intended to refer to.

**Comment by RizwanMunawar** [comment]
@Y-T-G @glenn-jocher ok I just updated the dog pose keypoints, list is available [here](https://learnopencv.com/animal-pose-estimation/#:~:text=Stanford%20Dogs%20Dataset%20for%20Animal%20Pose%20Estimation,-For%20our%20animal&text=The%20keypoint%20annotations%20are%20provided,%2C%20nose%2C%20and%20jaw) for reference.

- For `tiger-pose`, it is a very small dataset. When I was creating it, I didn't consider the keypoint names for each part; I just used indexes [0, 1,...] as names for each keypoint at that time. I can work on this sometime later, but for now, the ones @Y-T-G used look good to me, and all other datasets `kpts` names also look good to me.

**Review by RizwanMunawar — APPROVED** [review]

**Comment by glenn-jocher** [comment]
Great work, thanks @RizwanMunawar @Y-T-G!!

**Comment by UltralyticsAssistant** [comment]
🎉 Merged — amazing work on this, @Y-T-G, with valuable contributions from @glenn-jocher and @RizwanMunawar!

“It’s the little details that are vital. Little things make big things happen.” — John Wooden. Your addition of human-readable `kpt_names` turns indices into insight, making pose training, export, and inference more intuitive end-to-end. From clearer plots and logs to seamless deployment across backends (and Ultralytics HUB workflows), this is a meaningful quality-of-life boost for our community while staying fully backward compatible.

Thank you for raising the bar on clarity and usability. Onward to even more intuitive pose pipelines in YOLO11 and beyond!

See the details in PR #22411:
- [Human-readable keypoint names across training, export, and inference](https://github.com/ultralytics/ultralytics/pull/22411)
